### PR TITLE
Document IllegalArgumentException as throwable in EntityIdFormatter

### PR DESCRIPTION
**Pull request by AlaaSarhan**
Bug: https://phabricator.wikimedia.org/T207479

### CODE REVIEW

File: src/EntityId/EntityIdFormatter.php [hunk 1]
```diff
@@ -17,6 +17,7 @@ interface EntityIdFormatter {
 	 * @param EntityId $value
 	 *
 	 * @return string Either plain text, wikitext or HTML.
+	 * @throws \IllegalArgumentException
```
Comment (thiemowmde) [inline src/EntityId/EntityIdFormatter.php]: Can you please add half a sentence **when** this exception is thrown?

File: .travis.yml [hunk 2]
```diff
@@ -4,7 +4,7 @@ matrix:
   fast_finish: true
   include:
     - env: DM=~6.3
-      php: hhvm
+      php: hhvm-3.30
```
Comment (thiemowmde) [inline .travis.yml]: This change appears to be unrelated. It's not mentioned in the commit message or the Phabricator ticket.

Comment (AlaaSarhan) [reply .travis.yml]: It happens that current hhvm nightly does not support current version of composer (and soon enough not php). Travis was failing due to that.

This can be done in a separate PR, but think it's fine for this library to be done here since it's done in a separate commit.

Comment (thiemowmde) [reply .travis.yml]: Sure, I just wanted to have the reason documented. :-)

File: src/EntityId/EntityIdFormatter.php [hunk 3]
```diff
@@ -17,6 +17,7 @@ interface EntityIdFormatter {
 	 * @param EntityId $value
 	 *
 	 * @return string Either plain text, wikitext or HTML.
+	 * @throws \IllegalArgumentException when value falls out of supported patterns by the formatter
```
Comment (JeroenDeDauw) [inline src/EntityId/EntityIdFormatter.php]: Not on a dev machine so I can't check - should this not be InvalidArgumentException? That is what the phab ticket says

Comment (AlaaSarhan) [reply src/EntityId/EntityIdFormatter.php]: yes my bad .. fixed now
